### DISCUSSION TITLE
can't open pdf (The API version does not match the Worker version)

### DISCUSSION THREAD

**Discussion by mariosight** [RyotaUshio/obsidian-pdf-plus#125] [Q&A] [answered]
When I open the pdf, the console reports an error：The API version "3.9.0" does not match the Worker version "2.10.377"，Does it mean that my obsidian version is too low or too high? My current obsidian version is 1.5.11. Maybe I did something wrong about this, and what should I do. I'd be grateful if I could get some help.
![图片](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/81169033/b974d656-68d1-4bda-97ba-610caf05c0fb)

**Comment by RyotaUshio** [comment]
I really have no idea about this error. Does it happen only when PDF++ is enabled?

**Reply by mariosight** [reply]
Oh! I'm so sorry, this shouldn't be a problem with PDF++. This error also popped up after I closed the PDF++ plug-in. And when I recreate a new repository with only PDF++, it works fine. There is a problem with other things.
Sorry for disturbing you, and thanks for answering me!

**Reply by RyotaUshio** [reply]
No problem at all. My rough guess is that your Obsidian **installer** version might be too old (I'm not sure though).

From the trouble shooting guide on the [forum](https://forum.obsidian.md/t/about-the-bug-reports-category/24/11):

> If something looks wrong, first make sure you are using the latest Obsidian version and that **your installer is up-to-date. The app and installer versions don’t need to match, but they shouldn’t be too far apart**.
> The installer update needs to be done manually. Download and reinstall obsidian from the [website](https://obsidian.md/download).

**Reply by RyotaUshio** [reply, marked as answer]
I found this on the forum. It might be due to some other plugins loading a different PDF.js version.

https://forum.obsidian.md/t/cannot-load-pdf-with-the-new-pdf-viewer-api-version/61051/2

**Reply by mariosight** [reply]
Thank you so much!!! I have solved the problem by following the troubleshooting guide, and now it works fine! Next time I will follow those steps and try to figure out by myself. Thanks again!

**Reply by RyotaUshio** [reply]
glad it worked out for you. Can I ask what plugin was causing it?

**Reply by mariosight** [reply]
To be honest, I'm not sure...This problem disappeared when I turned on the safe mode, and then I canceled the plug-in from the beginning. I thought it might be the BookNote plug-in or DB folder plug-in? I turned off those plug-ins, but I am not sure if these plug-ins were the cause. Because when I open those plug-ins after reading pdf normally, it doesn't seem to affect it.

**Reply by mariosight** [reply]
Okay, I found it. It was caused by the BookNote plug-in. I just turned it on and then restarted Obsidian, and the problem occurred again. And I just found out that he is no longer updating in the plug-in market.

**Reply by RyotaUshio** [reply]
Great! I will add a notice about this somewhere in the README or the docs. Thanks!

**Reply by mariosight** [reply]
My pleasure! I'm glad this is my own problem and not PDF++'s problem, and thank you for providing this useful pdf reader!
